File: Interview Cake/19. Implement Queue With Two Stacks/queue.cpp
#include <stack>
#include <exception>
#include <stdexcept>
#include <iostream>

using namespace std;

class MyQueue {
	stack<int> pushStk;
	stack<int> popStk;

public: 
	void enqueue(int x) {
		pushStk.push(x);
	}

	int dequeue() {
		if (popStk.empty()) {
			if (pushStk.empty()) {
				throw runtime_error("empty queue");
			}

			while (!pushStk.empty()) {
				int topVal = pushStk.top();
				pushStk.pop();
				popStk.push(topVal);
			}
		}

		int topVal = popStk.top();
		popStk.pop();
		return topVal;
	}
};

int main() {
	MyQueue q = MyQueue();

	q.dequeue();
}

File: Interview Cake/19. Implement Queue With Two Stacks/queue.py
class Queue:
	def __init__(self):
		self.push_stk = []
		self.pop_stk = []

	def enqueue(self, x): 
		self.push_stk.append(x)

	def dequeue(self):
		if self.pop_stk:
			return self.pop_stk.pop()
		elif self.push_stk:
			while self.push_stk:
				self.pop_stk.append(self.push_stk.pop())
			return self.pop_stk.pop()
		else:
			raise Exception("Error: cannot pop empty queue.")